oem/ibm/libpldmresponder/file_io_type_smsmenu.cpp

```diff
@@ -0,0 +1,170 @@
+    std::string newPassStr;
+
+    // Format of file type USER_PASSWORD_AUTHENTICATION
+    // <User Name Length><User Name><Password Length><Password>
+
+    // Format of file type USER_PASSWORD_CHANGE
+    // <User Name Length><User Name><Old Password Length><Old Password>
+    // <New Password Length><New Password>
+
+    // Read the size of User Name which is the content of first four bytes
+    userNameLen = readVecContent(smsBuf, 0, sizeof(uint32_t));
+
+    // Read the size of Password/Old Password which is the content of four bytes
+    // after User Name
+    userPassLen =
+        readVecContent(smsBuf, sizeof(uint32_t) + userNameLen,
+                       sizeof(uint32_t) + userNameLen + sizeof(uint32_t));
+    if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_CHANGE)
+    {
+        // Read the size of New Password which is the content of four bytes
+        // after Old Password
+        userNewPassLen = readVecContent(
+            smsBuf,
+            sizeof(uint32_t) + userNameLen + sizeof(uint32_t) + userPassLen,
+            sizeof(uint32_t) + userNameLen + sizeof(uint32_t) + userPassLen +
+                sizeof(uint32_t));
+    }
+
+    // Split the vector to retrieve the User Name
+    result = vecSplit(smsBuf, sizeof(uint32_t), sizeof(uint32_t) + userNameLen);
+    std::string unameStr(result.begin(), result.end());
+
+    // Split the vector to retrieve the Password/Old Password
+    result = vecSplit(smsBuf, sizeof(uint32_t) + userNameLen + sizeof(uint32_t),
+                      sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                          userPassLen);
+    std::string passStr(result.begin(), result.end());
+
+    if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_CHANGE)
+    {
+        // Split the vector to retrieve the New Password
+        result = vecSplit(smsBuf,
+                          sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                              userPassLen + sizeof(uint32_t),
+                          sizeof(uint32_t) + userNameLen + sizeof(uint32_t) +
+                              userPassLen + sizeof(uint32_t) + userNewPassLen);
+        std::string tempNewPassStr(result.begin(), result.end());
+        newPassStr = tempNewPassStr;
+    }
+
+    metaDataObj.fileStatus = 0x00;
+    metaDataObj.fileMetaData1 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData2 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData3 = 0xFFFFFFFF;
+    metaDataObj.fileMetaData4 = 0xFFFFFFFF;
+
+    bool authenticated = false;
+    bool passwordChangeRequired = false;
+    bool operationAllowed = false;
+    changePasswordReasonCode changePassRc{};
+
+    if (smsMenuType == PLDM_FILE_TYPE_USER_PASSWORD_AUTHENTICATION)
+    {
+        authenticate(unameStr, passStr, authenticated, passwordChangeRequired,
+                     operationAllowed);
+        if (authenticated == true)
+        {
+            metaDataObj.fileMetaData1 = PasswordAuthResult::AUTHENTICATED;
+        }
+        else
```

Comment:
can we use conditional ternary operators?

Reply:
Done